File: 409-longest-palindrome/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int singleDigit=0;
        Map<Character,Integer>map= new HashMap<>();
        for(char c:s.toCharArray()){
            map.put(c,map.getOrDefault(c,0)+1);
            if(map.get(c)%2==1) singleDigit++;
            else singleDigit--;
        }
        return singleDigit>1?s.length()-singleDigit+1:s.length();
    }
}

File: 409-longest-palindrome/longest-palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        singleDigit=0
        mp={}
        for i in s:
            mp[i]=mp.get(i,0)+1
            if(mp[i]%2==1):
                singleDigit+=1
            else:
                singleDigit-=1
        return len(s)-singleDigit+1 if singleDigit>1 else len(s)